mysql-plugin/src/test/java/io/cdap/plugin/mysql/MysqlSinkTestRun.java

```diff
@@ -31,34 +29,84 @@
+    "dbRecord",
+    Schema.Field.of("ID", Schema.of(Schema.Type.INT)),
+    Schema.Field.of("NAME", Schema.of(Schema.Type.STRING)),
+    Schema.Field.of("SCORE", Schema.of(Schema.Type.DOUBLE)),
+    Schema.Field.of("GRADUATED", Schema.of(Schema.Type.BOOLEAN)),
+    Schema.Field.of("TINY", Schema.of(Schema.Type.INT)),
+    Schema.Field.of("SMALL", Schema.of(Schema.Type.INT)),
+    Schema.Field.of("BIG", Schema.of(Schema.Type.LONG)),
+    Schema.Field.of("MEDIUMINT_COL", Schema.of(Schema.Type.INT)),
+    Schema.Field.of("FLOAT_COL", Schema.of(Schema.Type.FLOAT)),
+    Schema.Field.of("REAL_COL", Schema.of(Schema.Type.DOUBLE)),
+    Schema.Field.of("NUMERIC_COL", Schema.decimalOf(PRECISION, SCALE)),
+    Schema.Field.of("DECIMAL_COL", Schema.decimalOf(PRECISION, SCALE)),
+    Schema.Field.of("BIT_COL", Schema.of(Schema.Type.BOOLEAN)),
+    Schema.Field.of("DATE_COL", Schema.of(Schema.LogicalType.DATE)),
+    Schema.Field.of("TIME_COL", Schema.of(Schema.LogicalType.TIME_MICROS)),
+    Schema.Field.of("TIMESTAMP_COL", Schema.of(Schema.LogicalType.TIMESTAMP_MICROS)),
+    Schema.Field.of("DATETIME_COL", Schema.of(Schema.LogicalType.TIMESTAMP_MICROS)),
+    Schema.Field.of("YEAR_COL", Schema.of(Schema.LogicalType.DATE)),
+    Schema.Field.of("TEXT_COL", Schema.of(Schema.Type.STRING)),
+    Schema.Field.of("TINYTEXT_COL", Schema.of(Schema.Type.STRING)),
+    Schema.Field.of("MEDIUMTEXT_COL", Schema.of(Schema.Type.STRING)),
+    Schema.Field.of("LONGTEXT_COL", Schema.of(Schema.Type.STRING)),
+    Schema.Field.of("CHAR_COL", Schema.of(Schema.Type.STRING)),
+    Schema.Field.of("BINARY_COL", Schema.of(Schema.Type.BYTES)),
+    Schema.Field.of("VARBINARY_COL", Schema.of(Schema.Type.BYTES)),
+    Schema.Field.of("TINYBLOB_COL", Schema.of(Schema.Type.BYTES)),
+    Schema.Field.of("BLOB_COL", Schema.of(Schema.Type.BYTES)),
+    Schema.Field.of("MEDIUMBLOB_COL", Schema.of(Schema.Type.BYTES)),
+    Schema.Field.of("LONGBLOB_COL", Schema.of(Schema.Type.BYTES)),
+    Schema.Field.of("ENUM_COL", Schema.of(Schema.Type.STRING)),
+    Schema.Field.of("SET_COL", Schema.of(Schema.Type.STRING))
+  );
+
+  private static final List<StructuredRecord> INPUT_RECORDS = new ArrayList<>();
+
+  @Before
+  public void setup() throws Exception {
+    try (Statement stmt = createConnection().createStatement()) {
+      stmt.execute("TRUNCATE TABLE MY_DEST_TABLE");
+    }
+  }
+
+  @Test
+  public void testDBSinkWithExplicitInputSchema() throws Exception {
+    testDBSink("testDBSinkWithExplicitInputSchema", "input-dbsinktest-explicit", true);
+  }
+
   @Test
-  public void testDBSink() throws Exception {
-    String inputDatasetName = "input-dbsinktest";
+  public void testDBSinkWithInferredInputSchema() throws Exception {
+    testDBSink("testDBSinkWithInferredInputSchema", "input-dbsinktest-inferred", false);
+  }
+
+  private void testDBSink(String appName, String inputDatasetName, boolean setInputSchema) throws Exception {
```

Comment:
avoid boolean params (better have 2 methods or schema as parameter, it will still allow to reuse same amount of code)

Reply:
Changed to pass schema as parameter.